pkg/apis/postgres-operator.crunchydata.com/v1beta1/postgrescluster_types.go

```diff
@@ -478,6 +478,23 @@ type PostgresInstanceSetSpec struct {
 	// More info: https://www.postgresql.org/docs/current/wal.html
 	// +optional
 	WALVolumeClaimSpec *corev1.PersistentVolumeClaimSpec `json:"walVolumeClaimSpec,omitempty"`
+
+	// The list of tablespaces volumes to mount for this postgrescluster
+	// This function is currently in alpha as of PGO v5.4
+	// +optional
+	TablespaceVolumes []TablespaceVolume `json:"tablespaceVolumes,omitempty"`
+}
+
+type TablespaceVolume struct {
+	// The name for the tablespace, used as the path name for the volume.
+	// Must be unique in the instance set since they become the directory names.
```

Comment:
🔧 Kubernetes can enforce uniqueness for us. Add `listType=map` and `listMapKey=name` on the `TablespaceVolumes` field.

https://kubernetes.io/docs/reference/using-api/server-side-apply/#merge-strategy

https://github.com/CrunchyData/postgres-operator/blob/8b14582aaa9df139b8eb38fe48a47f546db61f02/pkg/apis/postgres-operator.crunchydata.com/v1beta1/postgrescluster_types.go#L97-L98

Reply:
Added